editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -196,18 +196,66 @@ public boolean isProjectSpringBoot(IProject project) {
 		return CamelCatalogUtils.hasSpringBootDependency(projectDependencies);
 	}
 
+	
+	private org.fusesource.ide.camel.model.service.core.catalog.Dependency makeDependency(
+			String grpId, String artId, String version) {
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency dep = 
+				new org.fusesource.ide.camel.model.service.core.catalog.Dependency();
+		dep.setArtifactId(artId);
+		dep.setGroupId(grpId);
+		dep.setVersion(version);
+		return dep;
+	}
+	
 	/**
-	 * Add JAX-RS dependency
+	 * Add required dependencies
 	 */
 	private void updateDependencies(IProgressMonitor monitor) {
-		List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> deps = new ArrayList<>();
-		org.fusesource.ide.camel.model.service.core.catalog.Dependency one = 
-				new org.fusesource.ide.camel.model.service.core.catalog.Dependency();
-		one.setArtifactId("jboss-jaxrs-api_2.0_spec"); //$NON-NLS-1$
-		one.setGroupId("org.jboss.spec.javax.ws.rs"); //$NON-NLS-1$
-		one.setVersion("1.0.0.Final-redhat-1"); //$NON-NLS-1$
-		deps.add(one);
 		IProject project = options.getProject();
+		// use project to determine if we are building a spring or blueprint project
+		boolean isBlueprint = isProjectBlueprint(project);
+		boolean isSpringBoot = isProjectSpringBoot(project);
+
+		List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> deps = new ArrayList<>();
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency jaxrsSpec = 
+				makeDependency("org.jboss.spec.javax.ws.rs",  //$NON-NLS-1$
+						"jboss-jaxrs-api_2.0_spec",  //$NON-NLS-1$
+						"1.0.0.Final-redhat-1"); //$NON-NLS-1$
+		deps.add(jaxrsSpec);
+		
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency camelJackson = 
+				makeDependency("org.apache.camel",  //$NON-NLS-1$
+						"camel-jackson",  //$NON-NLS-1$
+						"${camel.version}"); //$NON-NLS-1$
```

Comment:
I think we can't rely on the fact that camel.version property is defined.

Reply:
Then what do you recommend? We also can't rely on the fact that the BOM is defined. Is there a way to get the version of the core camel dependency from the maven utilities?

Reply:
several cases are handled in org.fusesource.ide.projecttemplates.util.maven.MavenUtils.updateCamelVersionDependencies(Model, List<Dependency>, String) for instance.

To retrieve Camel version there is org.fusesource.ide.camel.model.service.core.util.CamelMavenUtils.getCamelVersionFromMaven(IProject, boolean)

Reply:
Perfect. Thanks